pytorch_lightning/utilities/parsing.py

```diff
@@ -186,15 +186,17 @@ def lightning_hasattr(model, attribute):
             attr = attribute in model.hparams
         else:
             attr = hasattr(model.hparams, attribute)
+    elif hasattr(model.datamodule, attribute):
```

Comment:
this is a bit confusing regarding the line above... mind adding a comment what is this case about...